### PR TITLE
7959a mapping

### PR DESCRIPTION
**Pull request by cloudmagic80**
## Summary

Initial form mapping for 7959a

### CODE REVIEW

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
```
Comment (balexandr) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:3]: Wrong module

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  class VHA107959A
+    include Virtus.model(nullify_blank: true)
+
+    attribute :data
+
+    def initialize(data)
+      @data = data
```
Comment (balexandr) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:10]: Need to add form_id and uuid

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    def metadata
+      {
+        'veteranFirstName' => @data.dig('veteran', 'full_name', 'first'),
+        'veteranLastName' => @data.dig('veteran', 'full_name', 'last'),
+        'fileNumber' => @data.dig('veteran', 'va_file_number').presence || @data.dig('veteran', 'ssn'),
+        'zipCode' => @data.dig('veteran', 'address', 'postal_code'),
+        'source' => 'VA Platform Digital Forms',
+        'docType' => @data['form_number'],
+        'businessLine' => 'CMP'
```
Comment (balexandr) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:21]: Need to add `'primaryContactInfo' => @data['primary_contact_info']`

Comment (balexandr) [reply modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:21]: I think there are also some other ones missing refer to other models as a reference

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb (hunk 1 above)
Comment (cloudmagic80) [reply modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:3]: good catch.

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    attribute :data
+    attr_reader :form_id
+
+    def initialize(data)
+      @data = data
+      @uuid = SecureRandom.uuid
+      @form_id = 'vha_10_7959a'
+    end
+
+    # rubocop:disable Lint/Syntax
```
Comment (balexandr) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:17]: I don't think this is necessary

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    def metadata
+      {
+        'veteranFirstName' => @data.dig('veteran', 'full_name', 'first'),
+        'veteranLastName' => @data.dig('veteran', 'full_name', 'last'),
+        'zipCode' => @data.dig('veteran', 'address', 'postal_code'),
+        'source' => 'VA Platform Digital Forms',
+        'docType' => @data['form_number'],
+        'businessLine' => 'CMP',
+        'country' => @data.dig('applicant_address', 'country') || 'USA',
```
Comment (balexandr) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:26]: Just for readability try to keep these matching across the models. `country` is under `zipCode` nitpick

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    def submission_date_config
+      {
+        should_stamp_date?: false,
+        page_number: 1,
+        title_coords: [440, 690],
+        text_coords: [440, 670]
+      }
+    end
+
+    # rubocop:disable Naming/BlockForwarding,Style/HashSyntax
```
Comment (balexandr) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:42]: Do we need this or can it just be
```
def method_missing(_, *args)
      args&.first
    end
```

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 7]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module IvcChampva
+  class VHA107959a
+    include Virtus.model(nullify_blank: true)
+    include Attachments
```
Comment (balexandr) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:6]: Does this form have attachments? IF not we don't need this include

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb (hunk 5 above)
Comment (cloudmagic80) [reply modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:25]: No. Not doing it

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb (hunk 4 above)
Comment (cloudmagic80) [reply modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb]: Yah, it wasn't required

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb (hunk 6 above)
Comment (cloudmagic80) [reply modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:40]: Without this, the forms won't attach

File: modules/ivc_champva/spec/models/vha_10_7979a_spec.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        'address' => { 'country' => 'USA', 'postal_code' => '12345' }
+      },
+      'form_number' => '10-7959A',
+      'veteran_supporting_documents' => [
+        { 'confirmation_code' => 'abc123' },
+        { 'confirmation_code' => 'def456' }
+      ]
+    }
+  end
+  let(:vha1010d) { described_class.new(data) }
```
Comment (balexandr) [inline modules/ivc_champva/spec/models/vha_10_7979a_spec.rb:27]: This is the wrong form number

**Review by balexandr — APPROVED** [review]

**Pull request merged**